stan/math/prim/mat/fun/gp_matern52_cov.hpp

```diff
@@ -96,18 +96,20 @@ gp_matern52_cov(const std::vector<T_x> &x, const T_s &sigma,
  * @tparam T_x type of elements contained in vector x
  * @tparam T_s type of element of sigma, the magnitude
  * @tparam T_l type of elements in vector of length scale
+ * @tparam R   num of rows in x matrix (either 1 or -1 since this is a vector)
```

Comment:
I always just say Eigen row type. Also use Eigen::Dynamic instead of -1. -1 is a weird magic constant, Eigen::Dynamic is something that can be Googled.

In Stan, this is either going to be Eigen::Dynamic or 1.

Also I don't think "since this is a vector" is confusing here. If it's a Stan vector, then R is always Eigen::Dynamic.

Reply:
My understanding was row vectors or vectors are Eigen::Matrix<T_x, 1, -1> or Eigen::Matrix<T_x, -1, 1>. I've seen this initialized else where in the library. 

> If it's a Stan vector, then R is always Eigen::Dynamic.

not always true. See all the test cases for `cov_exp_quad`, etc. This was implemented so that we can have `vector[D] x[N]` where this is a vector of column vectors or vectors of row vectors.

Also - quick question. If I specify templates with R, C the number of columns, does this help reduce the compile time? If so, I'll use this a default in order to reduce compile time for models that use these functions.

Reply:
We have `-1` in some code and `Eigen::Dynamic` in others and `Eigen::MatrixXd` in other still.  

I'm also happy to drop row vectors.  It's just a convenience feature in the sense that it avoids transposing in places where it's an input.  For example, we use row vectors, arrays, and vectors interchangeably to vectorize univariate probabilty functions.

Reply:
@drezap Let's drop RowVector-like things from the new GP covariance functions.

For the input variables for the GPs, let's roll with:
```
std::vector<double>
std::vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>
```

Is that okay? Otherwise there's just lots of extra annoying complexity in docs, function templating, and tests.

Reply:
That's OK with me.  I prefer to start with things we know we want then add other instances later if it turns out it's really inconvenient to work with the standard forms.  For example, `multi_normal` only accepts vectors, not row vectors.